drivers/gpio/sabertooth_analog_driver.go

```diff
@@ -0,0 +1,294 @@
+func (d SabertoothDIPSwitches) Bidirectional() bool { return d.DIPSwitch6() }
+
+// Unidirectional returns true if the DIP switch 6 is OFF
+func (d SabertoothDIPSwitches) Unidirectional() bool { return !d.DIPSwitch6() }
+
+type sabertoothAnalogDriver struct {
+    name         string
+    switches     SabertoothDIPSwitches
+    s1           string
+    s2           string
+    a1           string
+    a2           string
+    throttle     float64
+    differential float64
+    
+    // Used for differential change
+    m1Ratio, m2Ratio float64
+    
+    connection PwmWriter
+    gobot.Commander
+}
+
+// SabertoothAnalogDriver represents a sabertooth driver in analog mode
+type SabertoothAnalogDriver interface {
+    gobot.Driver
+    gobot.Commander
+    
+    // PinS1 returns the pin connected to the S1 terminal of the motor driver.
+    PinS1() string
+    
+    // PinS2 returns the pin connected to the S2 terminal of the motor driver.
+    PinS2() string
+    
+    // PinA1 returns the pin connected to the A1 terminal of the motor driver.
+    PinA1() string
+    
+    // PinA2 returns the pin connected to the A2 terminal of the motor driver.
+    PinA2() string
+    
+    // Throttle sets the throttle of both motors.
+    // The throttle value ranges from -1 to 1.
+    //
+    // A positive value turns the motors clockwise and a negative value turns the motors
+    // counterclockwise or the opposite depending on the motors connection.
+    // A value of 0 stops the motors.
+    Throttle(throttle float64) error
+    
+    // GetThrottle returns the current throttle value.
+    GetThrottle() float64
+    
+    // Differential sets the throttle difference between the two motors.
+    // The differential value ranges from -1 to 1.
+    //
+    // A negative value applies a lower power to the motor 1 and a positive value applies
+    // a lower power to the motor 2. A value of 0 applies the same power for both motors.
+    Differential(differential float64) error
+    
+    // GetDifferential returns the current differential value.
+    GetDifferential() float64
+    
+    // SetRampRate sets the speed ramping rate of both motors.
+    //
+    // NOTE: this operation is possible in bidirectional mode only.
+    SetRampRate(rate float64) error
+    
+    // SetMaxPower sets the maximum output for both motors.
+    //
+    // NOTE: this operation is possible in bidirectional mode only.
+    SetMaxPower(ratio float64) error
+}
+
+// NewSabertoothAnalogDriver creates a new SabertoothAnalogDriver given a PwmWriter,
+// the DIP switch settings and the pins connected to the S1, S2, A1 and A2 terminals
+// of the driver.
+//
+// Adds the following API commands:
+//  throttle(value float64) - see SabertoothAnalogDriver.Throttle
+//  differential(value float64) - see SabertoothAnalogDriver.Differential
+func NewSabertoothAnalogDriver(w PwmWriter, d SabertoothDIPSwitches, s1, s2, a1, a2 string) SabertoothAnalogDriver {
```

Comment:
Please could you add the link to the provided references in your commit message here in the code. This will simplify future development. You can have a look to i2c drivers for some examples.

A link to an reference code in another programming language is also helpful for future error catching, e.g. for i2c drivers there are often drivers already implemented in python or for arduino in C/C++.